pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl

```diff
@@ -9,11 +9,23 @@ The parameter `recursion_dict` is meant to preseve egality
 of converted objects and should never be given by the user.
 """
 
+## Default for conversion:
+## Do not convert Julia types, as long as type matches
+## If you have a GAP list of Julia Tuples, this function
+## enables conversion of that list to a list of type
+## Array{Tuple,1}.
+function gap_to_julia(t::T, x::Any) where T <: Type
+    if ! (typeof(x) <: t)
+        throw(MethodError("Wrong type: expected " * string(t) * ","))
+    end
+    return x
+end
+
 ## Default
 gap_to_julia(::Type{GAPInputType},x::GAPInputType) = x
 gap_to_julia(::Type{Any},         x::GAPInputType) = gap_to_julia(x)
```

Comment:
I was confused a bit by this rule; it really is different from the rules above and below it, as it performs some kind of "default conversion" for objects of type `GAPInputType`; I think it should be separated and a comment added. Of course it doesn't have to be in this PR, but I am pointing this out now as I had a fresh look at this conversion code, and think we really need to document it a bit better.

Reply:
Yeah, it indeed adds a default conversion (specially on `MPtr`) if the target type is `Any`. We need indeed to document it better, though.